### PR TITLE
implement `NOW()` siblings

### PR DESCRIPTION
**Pull request by jycor**
This PR has our behavior surrounding `NOW()` functions more closely match MySQL.

Changes:
 - Added `NOW()` synonyms to registry
 - Have `CURRENT_TIMESTAMP()`, `LOCALTIME()`, `LOCALTIMESTAMP()` all just call `NOW()`
 - Support parsing synonyms in `DEFAULT` and `ON UPDATE` expressions
 - Fixed `SHOW CREATE TABLE` to print `CURRENT_TIMESTAMP` for `NOW()` and synonyms

Companion PR: https://github.com/dolthub/vitess/pull/296
Fixes:
 - https://github.com/dolthub/dolt/issues/7129
 - https://github.com/dolthub/dolt/issues/6058

### CODE REVIEW

**Review by max-hoffman — APPROVED** [review]
LGTM, the biggest thing I'm noticing is that we could frontload some of the precision resolving. Not blocking though

File: sql/planbuilder/ddl.go [hunk 1]
```diff
@@ -1352,12 +1396,13 @@ func (b *Builder) convertDefaultExpression(inScope *scope, defaultExpr ast.Expr,
...
 		if _, ok := resExpr.(sql.FunctionExpression); ok {
 			switch resExpr.(type) {
-			case *function.Now, *function.CurrTimestamp:
+			case *function.Now:
 				// Datetime and Timestamp columns allow now and current_timestamp to not be enclosed in parens,
 				// but they still need to be treated as function expressions
 				isLiteral = false
 			default:
 				// All other functions must *always* be enclosed in parens
+				// TODO: remove this
```
[the remark this answers is not in the document]
Comment (jycor) [reply sql/planbuilder/ddl.go]: I initially wanted this to match MySQL exactly, but they just throw a generic syntax error. I think our error is more descriptive; removed the TODO

File: sql/columndefault.go [hunk 2]
```diff
@@ -147,7 +147,12 @@ func (e *ColumnDefaultValue) String() string {
 	if e.Literal {
 		return e.Expr.String()
 	} else {
-		return fmt.Sprintf("(%s)", e.Expr.String())
+		// There's a special case for NOW()
+		str := e.Expr.String()
```
[the remark this answers is not in the document]
Comment (jycor) [reply sql/columndefault.go]: `e.Expr.(*function.Now)` results in an import cycle

File: sql/expression/function/time.go [hunk 3]
```diff
@@ -910,50 +891,124 @@ func (*Now) CollationCoercibility(ctx *sql.Context) (collation sql.CollationID,
...
-	if !b {
+	// Should syntax error before this; check anyway
+	if prec == nil {
+		return nil, ErrTimeUnexpectedlyNil.New(n.FunctionName())
+	}
+
+	// Must receive integer
+	// Should syntax error before this; check anyway
+	var fsp int
+	switch p := prec.(type) {
```
[the remark this answers is not in the document]
Comment (jycor) [reply sql/expression/function/time.go]: It looks like there are other functions, like `CURRENT_TIME` and `UTC_TIMESTAMP` that should be using similar logic. Will do this refactor in a follow up PR.

**Pull request merged**